File: app/src/main/java/com/spotsencesdk/ui/activity/MainActivity.java
package com.spotsensesdk.ui.activity;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.spotsensesdk.R;
import com.spotsensesdk.utils.GlobalMethods;
import com.spotsense.interfaces.GetSpotSenseData;
import com.spotsense.utils.spotSenseGeofencing.SpotSense;


public class MainActivity extends AppCompatActivity implements GetSpotSenseData {


    private String ChannelId = "ChannelId";
    SpotSense spotSense;


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        AddSpotSenseGeo();

    }


    public void AddSpotSenseGeo() {
        spotSense = new SpotSense(this, "clientIds", "ClientSecret", MainActivity.this);//sagar ifuturz
        spotSense.start();

    }

    @Override
    public void getSpotSenseBeaconData(String beaconTransactions, String beaconName) {
        if (beaconTransactions.equalsIgnoreCase("Enter")) {
            GlobalMethods.sendNotification(MainActivity.this, 14, ChannelId, beaconName, "Welcome Zak using beacone", MainActivity.class, R.drawable.notification, R.drawable.notification);
        } else {
            GlobalMethods.sendNotification(MainActivity.this, 15, ChannelId, beaconName, "Visit Again Zak  using beacone", MainActivity.class, R.drawable.notification, R.drawable.notification);
        }

    }

    @Override
    public void getSpotSenseGeofencingData(String GeofenceTransactions, /*ArrayList<String> GeofenceTransactionsRequestedId,*/ String geofenceName) {

        if (GeofenceTransactions.equalsIgnoreCase("Exited")) {
            GlobalMethods.sendNotification(MainActivity.this, 13, ChannelId, geofenceName, "Visit Again Zak", MainActivity.class, R.drawable.notification, R.drawable.notification);
        } else {
            GlobalMethods.sendNotification(MainActivity.this, 12, ChannelId, geofenceName, "Welcome Zak", MainActivity.class, R.drawable.notification, R.drawable.notification);
        }

    }

}

File: app/src/main/java/com/app/spotsensesdk/ui/activity/MainActivity.kt
package com.app.spotsensesdk.ui.activity

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.app.spotsense.interfaces.GetSpotSenseData
import com.app.spotsense.utils.spotSenseGeofencing.SpotSense
import com.app.spotsensesdk.R
import com.app.spotsensesdk.utils.GlobalMethods

class MainActivity : AppCompatActivity(), GetSpotSenseData {
    private val channelID = "ChannelId"
    private var spotSense: SpotSense? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        addSpotSenseGeo()
    }

    private fun addSpotSenseGeo() {
//        spotSense = SpotSense(this, "clientIds", "ClientSecret", this@MainActivity) //sagar ifuturz
        spotSense = SpotSense(
            this,
            getString(R.string.testClientID),
            getString(R.string.testClientSecret),
            this@MainActivity
        )
        spotSense!!.start()
    }

    override fun getSpotSenseGeofencingData(geofenceTransactions: String?, geofenceName: String?) {
        if (geofenceTransactions.equals("Exited",ignoreCase = true)) GlobalMethods.sendNotification(
            this@MainActivity, 13, channelID, geofenceName, "Visit Again Zak",
            MainActivity::class.java, R.drawable.notification, R.drawable.notification
        ) else GlobalMethods.sendNotification(
            this@MainActivity, 12, channelID, geofenceName, "Welcome Zak",
            MainActivity::class.java, R.drawable.notification, R.drawable.notification
        )
    }

    override fun getSpotSenseBeaconData(beaconTransactions: String?, beaconName: String?) {
        if (beaconTransactions.equals("Enter", ignoreCase = true)) GlobalMethods.sendNotification(
            this@MainActivity, 14, channelID, beaconName, "Welcome Zak using beacone",
            MainActivity::class.java, R.drawable.notification, R.drawable.notification
        ) else GlobalMethods.sendNotification(
            this@MainActivity, 15, channelID, beaconName, "Visit Again Zak  using beacone",
            MainActivity::class.java, R.drawable.notification, R.drawable.notification
        )
    }
}